src/routes/ar-io.ts

```diff
@@ -133,6 +133,46 @@ arIoRouter.put('/ar-io/admin/block-data', express.json(), async (req, res) => {
   }
 });
 
+// Block resolution of ArNS name
+arIoRouter.put('/ar-io/admin/block-name', express.json(), async (req, res) => {
+  try {
+    const { name, source, notes } = req.body;
+    if (name === undefined) {
+      res.status(400).send("Must provide 'name'");
+      return;
+    }
+
+    system.db.blockName({ name, source, notes });
+    system.blockedNamesCache.addName(name);
+
+    res.json({ message: 'Name blocked' });
+  } catch (error: any) {
+    res.status(500).send(error?.message);
+  }
+});
+
+// Unblock resolution of ArNS name
+arIoRouter.put(
+  '/ar-io/admin/unblock-name',
+  express.json(),
+  async (req, res) => {
+    try {
+      const { name } = req.body;
+      if (name === undefined) {
+        res.status(400).send("Must provide 'name'");
+        return;
+      }
+
+      system.db.unblockName({ name });
+      system.blockedNamesCache.removeName(name);
```

Comment:
Should we just run `db.unblockName` from the cache itself and just call `removeName` here?

Reply:
I lean towards keeping them separate. That keeps the cache interface simpler.